src/main/java/org/apache/commons/lang3/Validate.java

```diff
@@ -0,0 +1,1329 @@
+     *
+     * <pre>Validate.exclusiveBetween(0, 2, 1);</pre>
+     *
+     * @param <T> the type of the argument object
+     * @param start  the exclusive start value, not null
+     * @param end  the exclusive end value, not null
+     * @param value  the object to validate, not null
+     * @throws IllegalArgumentException if the value falls outside the boundaries
+     * @see #exclusiveBetween(Object, Object, Comparable, String, Object...)
+     *
+     * @since 3.0
+     */
+    public static <T> void exclusiveBetween(final T start, final T end, final Comparable<T> value) {
+        // TODO when breaking BC, consider returning value
+        if (value.compareTo(start) <= 0 || value.compareTo(end) >= 0) {
+            throw new IllegalArgumentException(String.format(DEFAULT_EXCLUSIVE_BETWEEN_EX_MESSAGE, value, start, end));
+        }
+    }
+
+    /**
+     * <p>Validate that the specified argument object fall between the two
+     * exclusive values specified; otherwise, throws an exception with the
+     * specified message.</p>
+     *
+     * <pre>Validate.exclusiveBetween(0, 2, 1, "Not in boundaries");</pre>
+     *
+     * @param <T> the type of the argument object
+     * @param start  the exclusive start value, not null
+     * @param end  the exclusive end value, not null
+     * @param value  the object to validate, not null
+     * @param message  the {@link String#format(String, Object...)} exception message if invalid, not null
+     * @param values  the optional values for the formatted exception message, null array not recommended
+     * @throws IllegalArgumentException if the value falls outside the boundaries
+     * @see #exclusiveBetween(Object, Object, Comparable)
+     *
+     * @since 3.0
+     */
+    public static <T> void exclusiveBetween(final T start, final T end, final Comparable<T> value, final String message, final Object... values) {
+        // TODO when breaking BC, consider returning value
+        if (value.compareTo(start) <= 0 || value.compareTo(end) >= 0) {
+            throw new IllegalArgumentException(String.format(message, values));
+        }
+    }
+
+    /**
+     * Validate that the specified primitive value falls between the two
+     * exclusive values specified; otherwise, throws an exception.
+     *
+     * <pre>Validate.exclusiveBetween(0, 2, 1);</pre>
+     *
+     * @param start the exclusive start value
+     * @param end   the exclusive end value
+     * @param value the value to validate
+     * @throws IllegalArgumentException if the value falls out of the boundaries
+     *
+     * @since 3.3
+     */
+    @SuppressWarnings("boxing")
+    public static void exclusiveBetween(final long start, final long end, final long value) {
+        // TODO when breaking BC, consider returning value
+        if (value <= start || value >= end) {
+            throw new IllegalArgumentException(String.format(DEFAULT_EXCLUSIVE_BETWEEN_EX_MESSAGE, value, start, end));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format long: value (declared at line 1154), start (declared at line 1154), end (declared at line 1154). This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 1157. Consider formatting this data with &#39;%d&#39; instead.